broker/pkg/platform/kube/crd/types.go

```diff
@@ -0,0 +1,374 @@
+// Code generated by generate.sh. DO NOT EDIT!
+// Sources: pkg/model/config/resource.go
+// Output: pkg/platform/kube/crd/types.go
+
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crd
+
+// This file contains Go definitions for Custom Resource Definition kinds
+// to adhere to the idiomatic use of k8s API machinery.
+// These definitions are synthesized from Istio configuration type descriptors
+// as declared in the Broker config model.
+
+import (
```

Comment:
This file should not be checked in. We can use go-gen way, but we need to drop bazel first since the two ways cannot coexist.

Reply:
It is all content that is generated by bazel. 
When this file is not generated by bazel we will delete it from here, but until then this + glide gives us a workspace where bazel is not needed unless you want to use it.

Reply:
That's the thing. We have to use bazel to generate this and a few other files, which is a thing you're trying avoid. Let's wait until we ditch bazel fully, to replace generation scripts with bash or go-gen or something else. I'm ok committing this temporarily until we fully get rid of bazel.

Reply:
Is ditching bazel plan of record and/or documented somewhere?

Reply:
I learned it from @rshriram :) I don't think with the current trend of repo + monorepo it makes sense to use bazel anymore.

Reply:
yes I think dropping bazel is orthogonal to this.  This ensures that bazel is not __required__ if you do not muck with protos et al.